applications/serial_lte_modem/src/main.c

```diff
@@ -366,6 +367,46 @@ void handle_mcuboot_swap_ret(void)
 	}
 }
 
+int nidd_auto_connect(void)
+{
+	int err = 0;
+	int cfun;
+
+#if defined(CONFIG_LWM2M_CARRIER_SERVER_BINDING_N)
+	if (!fota_reconnect) {
+		return 0;
+	}
+	err = nrf_modem_at_scanf("AT+CFUN?", "+CFUN: %d", &cfun);
+	if (err != 1 || (cfun == 1 || cfun == 5)) {
+		return 0;
+	}
+
+	LOG_INF("NIDD auto connect");
+	err = nrf_modem_at_printf("AT%%XSYSTEMMODE=0,1,0,0");
+	if (err) {
+		LOG_ERR("Failed to configure RAT: %d", err);
+		return err;
+	}
+	err = nrf_modem_at_printf("AT+CGDCONT=0,\"Non-ip\"");
+	if (err) {
```

Comment:
This step is to be changed, as some NIDD network like Jio in India does requires APN.
So PDP Context configuration will have to be left to PDN init after modem init.